### ISSUE TITLE
ROS melodic, Gazebo version 9

### ISSUE THREAD

**Issue by MiPa12** [marinaKollmitz/gazebo_ros_2Dmap_plugin#3]
Hi,
I want to use this package on Ubuntu 18 with ROS melodic and gazebo version 9. After catkin_make I got the following error:
`
fatal error: gazebo/math/Vector3.hh: No such file or directory
 #include <gazebo/math/Vector3.hh>
`
which is explained here:
https://answers.gazebosim.org/question/21872/vector3hh-not-found-on-gazebo9/

So I changed math::Vector3 to ignition::math::Vector3
with the header #include <ignition/math/Vector3.hh>

but I still get some errors:
`
[  0%] Built target gazebo_ros_2Dmap_plugin_xacro_generated_to_devel_space_
[ 50%] Building CXX object gazebo_ros_2Dmap_plugin/CMakeFiles/gazebo_2Dmap_plugin.dir/src/gazebo_2Dmap_plugin.cpp.o
/home/user/map_ws/src/gazebo_ros_2Dmap_plugin/src/gazebo_2Dmap_plugin.cpp: In member function ‘virtual void gazebo::OccupancyMapFromWorld::Load(gazebo::physics::WorldPtr, sdf::ElementPtr)’:
/home/user/map_ws/src/gazebo_ros_2Dmap_plugin/src/gazebo_2Dmap_plugin.cpp:68:44: error: qualified-id in declaration before ‘(’ token
 bool OccupancyMapFromWorld::ServiceCallback(std_srvs::Empty::Request& req,
                                            ^
/home/user/map_ws/src/gazebo_ros_2Dmap_plugin/src/gazebo_2Dmap_plugin.cpp:75:50: error: qualified-id in declaration before ‘(’ token
 bool OccupancyMapFromWorld::worldCellIntersection(const ignition::math::Vector3& cell_center,
                                                  ^
/home/user/map_ws/src/gazebo_ros_2Dmap_plugin/src/gazebo_2Dmap_plugin.cpp:124:39: error: qualified-id in declaration before ‘(’ token
 void OccupancyMapFromWorld::cell2world(unsigned int cell_x, unsigned int cell_y,
                                       ^
/home/user/map_ws/src/gazebo_ros_2Dmap_plugin/src/gazebo_2Dmap_plugin.cpp:133:39: error: qualified-id in declaration before ‘(’ token
 void OccupancyMapFromWorld::world2cell(double world_x, double world_y,
                                       ^
/home/user/map_ws/src/gazebo_ros_2Dmap_plugin/src/gazebo_2Dmap_plugin.cpp:142:39: error: qualified-id in declaration before ‘(’ token
 bool OccupancyMapFromWorld::cell2index(int cell_x, int cell_y,
                                       ^
/home/user/map_ws/src/gazebo_ros_2Dmap_plugin/src/gazebo_2Dmap_plugin.cpp:158:39: error: qualified-id in declaration before ‘(’ token
 bool OccupancyMapFromWorld::index2cell(int index, unsigned int cell_size_x,
                                       ^
/home/user/map_ws/src/gazebo_ros_2Dmap_plugin/src/gazebo_2Dmap_plugin.cpp:174:47: error: qualified-id in declaration before ‘(’ token
 void OccupancyMapFromWorld::CreateOccupancyMap()
                                               ^
In file included from /usr/include/gazebo-9/gazebo/common/common.hh:36:0,
                 from /usr/include/gazebo-9/gazebo/gazebo_core.hh:19,
                 from /usr/include/gazebo-9/gazebo/gazebo.hh:20,
                 from /home/user/map_ws/src/gazebo_ros_2Dmap_plugin/include/common.h:26,
                 from /home/usermap_ws/src/gazebo_ros_2Dmap_plugin/include/gazebo_2Dmap_plugin.h:28,
                 from /home/user/map_ws/src/gazebo_ros_2Dmap_plugin/src/gazebo_2Dmap_plugin.cpp:21:
/home/user/map_ws/src/gazebo_ros_2Dmap_plugin/src/gazebo_2Dmap_plugin.cpp:283:1: error: expected unqualified-id before string constant
 GZ_REGISTER_WORLD_PLUGIN(OccupancyMapFromWorld)
 ^
/home/user/map_ws/src/gazebo_ros_2Dmap_plugin/src/gazebo_2Dmap_plugin.cpp:283:1: error: a function-definition is not allowed here before ‘{’ token
 GZ_REGISTER_WORLD_PLUGIN(OccupancyMapFromWorld)
 ^
`
any tip will be appreciated, thank you

**Comment by marinaKollmitz** [comment]
Hi, yes I think the problem is that you need to change math::Vector3 to ignition::math::Vector3**d**. I adapted the code for ros melodic and gazebo 9, please check out the new melodic-devel branch :-). I hope it works. Could you please confirm that the code works and then close the issue? Thank you, best! Marina

**Comment by MiPa12** [comment]
Hi, I tested it. It worked well, thank you very much.
The only thing I recognized is that after I insert the the plugin code into my world file, after launching I got with different worlds the following error. 
`Error [Element.cc:702] Missing element description for [contactSensor]`
This does not effect the generation of the yaml and pgm file but I'm not sure if this is also a problem of the switch from gazebo version 7 to 9

After roslaunch:
.....
`
[ INFO] [1590410604.017226244]: Physics dynamic reconfigure ready.
[ INFO] [1590410604.148032308]: occupancy map plugin started
Error [Element.cc:702] Missing element description for [contactSensor]
`

After rosservice call:

`[ INFO] [1590410604.017226244]: Physics dynamic reconfigure ready.
[ INFO] [1590410604.148032308]: occupancy map plugin started
Error [Element.cc:702] Missing element description for [contactSensor]
Starting wavefront expansion for mapping
Occupancy Map generation completed
`

best, Michael
